py/selenium/deprecated.py

```diff
@@ -0,0 +1,31 @@
+from warnings import warn
+
+removed_from_version = "4.12"
+
+
+def deprecated_function(message):
+    """decorator to log deprecation warning messgaes for deprecated methods."""
+
+    def _deprecated_function(func):
+        def wrapper(*args, **kwargs):
+            warn(f"{message}: will be removed from {removed_from_version}", DeprecationWarning, stacklevel=2)
+            return func(*args, **kwargs)
+
+        return wrapper
+
+    return _deprecated_function
+
+
+def deprecated_attributes(*attrs, message):
+    def _deprecated_attributes(func):
+        def wrapper(*args, **kwargs):
+            result = func(*args, **kwargs)
+            for attr in attrs:
+                _attr = getattr(args[0], attr, None)
+                if _attr:
```

Comment:
not sure this work as you are expecting, e.g a deprecated attribute that been assigned `falsy`, e.g `log_path=False` would probably not warn here?

Reply:
I have tweaked the logic. Now we can pass arbitrary keyword argument to the decorator where we can specify "should the user be warned if the deprecated attribute is falsy or should the user be warned if the he tries to pass some value to deprecated attribute which evaluates to truthy"